### PR TITLE
Separate models from examples/tests; benchmarking script

### PR DESCRIPTION
**Pull request by ae-foster**
This PR does several things:
 - allow unknown covariance in linear models and introduce the normal inverse gamma guide family for inference
 - reorganize the code to separate models from examples
 - add a benchmarking script - right now comparing VI against exact answers only

### CODE REVIEW

File: examples/contrib/oed/eig_estimation_benchmarking.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
+import time
```
Comment (martinjankowiak) [inline examples/contrib/oed/eig_estimation_benchmarking.py]: is this supposed to work? when i run this i get "UserWarning: Encountered NaN: loss"

Comment (ae-foster) [reply examples/contrib/oed/eig_estimation_benchmarking.py]: Yeah I saw this once before. What happens if you run it again?

Comment (martinjankowiak) [reply examples/contrib/oed/eig_estimation_benchmarking.py]: ah, seems to be a pytorch 0.4.1 versus 0.4 problem.

@neerajprad @fritzo if you're looking for 0.4.1/0.4 discrepancies try

pytest -s -v examples/contrib/oed/eig_estimation_benchmarking.py 

works fine in 0.4 but i get "UserWarning: Encountered NaN: loss" in 0.4.1

no idea why

Comment (ae-foster) [reply examples/contrib/oed/eig_estimation_benchmarking.py]: Hmmm... I'll stick to 0.4 for local testing for now

**Pull request closed without merging**